File: blog/snippets/python/selection_sort.py
# Selection sort algorithm
""" The code is purposefully not written using Pythonic idioms """

def selection_sort(array):
    for i in range(0, len(array)):
        min_index = i
        for j in range(i + 1, len(array)):
            if array[j] < array[min_index]:
                min_index = j
        array[i], array[min_index] = array[min_index], array[i]
    return array


random_array = [13, 25, 26, 17, 48, 78, 34, 78, 12, 56, 89, 45]

sorted_list = selection_sort(random_array)
print(f'The sorted list is {sorted_list}')

File: blog/snippets/javascript/selection_sort.js
// Selection sort algorithm

function selectionSort(array) {
  for (var i = 0; i < array.length; i++) {
    var minIndex = i;
    for (var j = i + 1; j < array.length; j++) {
      if(array[j] < array[minIndex]) {
        minIndex = j;
      }
    }
    if (minIndex != i) {
      var temp = array[i];
      array[i] = array[minIndex];
      array[minIndex] = temp;
    }
  }
  return array;
}

randomArray = [13,25,26,17,48,78,34,78,12,56,89,45]

sortedList = selectionSort(randomArray);
console.log(`The sorted list is ${sortedList}`);
